get-artifacts/get_artifacts.sh

```diff
@@ -0,0 +1,168 @@
+#!/bin/bash
+function echoerr() {
+  cat <<< "$@" 1>&2;
+}
+
+#Check prerequists.
+function check_command() {
+    for var in "$@"; do
+      if ! (command -v "$var" >/dev/null 2>&1); then
+          echoerr "This command requires $var to be installed"
+          exit 1
+      fi
+    done
+}
+check_command jq curl zip echo getopts
+
+function usage() {
+  echoerr -t token used to communicate with github actions.
+  echoerr -h number of workflow_runs to pull artifacts from.
+  echoerr -a space seperated list of artifact names.
+  echoerr -b optional branch to pull historical artifacts from.
+  echoerr -w workflow file name of of workflow to pull artifact from.
+  echoerr -d target directory where subdirectories for jobs will be created and artifacts will be unzip to.
+  echoerr -i the single job to artifacts from.  Overrides -b,-w, and -h
+  echoerr -z decompress and delete the original downloaded artifacts?
+  echoerr -? this message.
+  echoerr output will written in to the -t target directory.
+}
+
+TOKEN=
+HISTORY=
+ARTIFACTS=
+REPO=
+WORKFLOW=
+BRANCH=
+TARGET_DIR=
+WORKFLOW_RUN_ID=
+DECOMPRESS=false
+
+
+while getopts 'h:a:r:w:b:i:t:d:z' OPTION; do
```

Comment:
bleh, only having access to single-character options is one of the reasons I really wish this weren't in bash :(